awx/main/models/unified_jobs.py

```diff
@@ -810,11 +817,18 @@ def save(self, *args, **kwargs):
 
         # Sanity check: Has the job just completed? If so, mark down its
         # completion time, and record its output to the database.
+        #import sdb
+        #sdb.set_trace()
         if self.status in ('successful', 'failed', 'error', 'canceled') and not self.finished:
-            # Record the `finished` time.
+            # Record the `finished` time and if the job was canceled, set it to null.
             self.finished = now()
+            if self.status == 'canceled':
+                self.finished = None
```

Comment:
If it reports being both finished and canceled, I'd rather know than to lose that information here.

Reply:
this is starting to make sense to me, in that there are some cases where we should desire to not set `finished`.

However, I would try to do that in a more targeted way. Specifically, if `started` datetime field is None, and status is canceled, then `finished` should not be set.